asdf/generic_io.py

```diff
@@ -754,7 +754,7 @@ def __init__(self, fd, mode, close=False, uri=None):
         super().__init__(fd, mode, close=close, uri=uri)
 
         if uri is None and hasattr(fd, "name") and isinstance(fd.name, str):
-            self._uri = util.filepath_to_url(os.path.abspath(fd.name))
+            self._uri = pathlib.Path(fd.name).absolute().as_uri()
```

Comment:
can `self._uri` handle `~`? If not, add `.expanduser()`?
```suggestion
            self._uri = pathlib.Path(fd.name).expanduser().absolute().as_uri()
```

Reply:
Nice! So many pathlib features :)